src/network/networkd-bridge-mdb.c

```diff
@@ -125,24 +125,35 @@ static int bridge_mdb_configure(BridgeMDB *mdb, Link *link, Request *req) {
                                IN_ADDR_TO_STRING(mdb->family, &mdb->group_addr), mdb->vlan_id);
 
         entry = (struct br_mdb_entry) {
-                /* If MDB entry is added on bridge master, then the state must be MDB_TEMPORARY.
+                /* If MDB entry is added on bridge master, then the state must be MDB_TEMPORARY,
+                 * except on L2 routes, where they must always be permanent.
                  * See br_mdb_add_group() in net/bridge/br_mdb.c of kernel. */
-                .state = link->master_ifindex <= 0 ? MDB_TEMPORARY : MDB_PERMANENT,
+                .state = (link->master_ifindex <= 0) && (mdb->type == BRIDGE_MDB_ENTRY_TYPE_L3) ? MDB_TEMPORARY : MDB_PERMANENT,
```

Comment:
the () are unnecessary

Reply:
I had to review the C operators priority table for this one, and I still think it's easier to read with the parentheses, but you're right, they're redundant. Removed!